### PR TITLE
FAC-798: Implement DeleteAdviceItem Service

### PR DESCRIPTION
**Pull request by maziyar-gerami**
### Checklist
Before you label your PR as 'Ready To Review', please check the below list and put an 'X' char into each bracket;

[X] - Did you update the Postman file?
[X] - Did you update the corresponding document?
[X] - Did you review the PR by yourself?
[X] - Did you merge the main branch into this branch recently?
[X] - Did you remove the probable extra code?
[X] - Did you label the PR as 'Ready to Review'?
[X] - Did you update the Jira task to 'In Review'?
[X] - Did you add the Confluence's link to the Jira task as a description?

### CODE REVIEW

File: flickit-advice-engine/src/main/java/org/flickit/assessment/advice/adapter/in/rest/adviceitem/DeleteAdviceItemRestController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+@RestController
+@RequiredArgsConstructor
+public class DeleteAdviceItemRestController {
+
+    private final UserContext userContext;
+    private final DeleteAdviceItemUseCase useCase;
+
+    @DeleteMapping("/advice-items/{adviceItemId}")
+    ResponseEntity<Void> createAdviceItem(@PathVariable("adviceItemId") UUID adviceItemId) {
```
Comment (mhzkh) [inline flickit-advice-engine/src/main/java/org/flickit/assessment/advice/adapter/in/rest/adviceitem/DeleteAdviceItemRestController.java:21]: rename to `deleteAdviceItem`

File: flickit-advice-engine/src/main/java/org/flickit/assessment/advice/adapter/in/rest/adviceitem/DeleteAdviceItemRestController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        UUID currentUserId = userContext.getUser().id();
+        useCase.deleteAdviceItem(toParam(adviceItemId, currentUserId));
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    private Param toParam(UUID adviceItemId, UUID currentUserId) {
+        return new Param(adviceItemId,
```
Comment (mhzkh) [inline flickit-advice-engine/src/main/java/org/flickit/assessment/advice/adapter/in/rest/adviceitem/DeleteAdviceItemRestController.java:28]: Can be a single line

File: flickit-advice-engine/src/test/java/org/flickit/assessment/advice/application/service/adviceitem/DeleteAdviceItemServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        when(loadAdviceItemPort.loadAdviceItem(param.getAdviceItemId())).thenReturn(Optional.empty());
+
+        var throwable = assertThrows(ResourceNotFoundException.class, () -> service.deleteAdviceItem(param));
+        assertEquals(DELETE_ADVICE_ITEM_ADVICE_ITEM_NOT_FOUND, throwable.getMessage());
+
+        verifyNoInteractions(loadAssessmentResultPort, assessmentAccessChecker, deleteAdviceItemPort);
+    }
+
+    @Test
+    void testDeleteAdviceItem_whenAssessmentResultDoesNotExist_thenThrowResourceNotFoundException() {
```
Comment (mhzkh) [inline flickit-advice-engine/src/test/java/org/flickit/assessment/advice/application/service/adviceitem/DeleteAdviceItemServiceTest.java:60]: تو عنوان تست بالایی از عبارت `NotExists` استفاده کردید. اینجا از عبارت `DoesNotExist` استفاده شده.
لطفا حداقل تو یه کلاس تست، تست‌ها رو با یه قاعده یکسان نام‌گذاری کنید.

File: flickit-advice-engine/src/test/java/org/flickit/assessment/advice/application/service/adviceitem/DeleteAdviceItemServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        when(loadAdviceItemPort.loadAdviceItem(param.getAdviceItemId())).thenReturn(Optional.empty());
+
+        var throwable = assertThrows(ResourceNotFoundException.class, () -> service.deleteAdviceItem(param));
+        assertEquals(DELETE_ADVICE_ITEM_ADVICE_ITEM_NOT_FOUND, throwable.getMessage());
+
+        verifyNoInteractions(loadAssessmentResultPort, assessmentAccessChecker, deleteAdviceItemPort);
+    }
+
+    @Test
+    void testDeleteAdviceItem_whenAssessmentResultNotExist_thenThrowResourceNotFoundException() {
```
Comment (m-khorsandi) [inline flickit-advice-engine/src/test/java/org/flickit/assessment/advice/application/service/adviceitem/DeleteAdviceItemServiceTest.java:60]: Change `NotExist` to `NotExists`

Comment (m-khorsandi) [reply flickit-advice-engine/src/test/java/org/flickit/assessment/advice/application/service/adviceitem/DeleteAdviceItemServiceTest.java:60]: done

**Pull request merged**